internal/app/xhttp/xhttp.go

```diff
@@ -10,10 +11,21 @@ func New(config conf.Config) *echo.Echo {
 	srv := echo.New()
 	srv.HideBanner = true
 	srv.HidePort = true
+
+	if config.EnableAuthentication() {
+		srv.Use(middleware.BasicAuth(func(username, password string, c echo.Context) (bool, error) {
+			if username == config.AuthenticationUsername() && password == config.AuthenticationPassword() {
```

Comment:
FYI @odarriba @gulien at a glance it appears that this is vulnerable to timing attacks because you're using plain `==` to compare credentials rather than using a secure constant time comparison operation.

Reply:
You are right. In theory, if the strings to compare are strong and the network conditions remains stable, one could really measure when a username is really valid.

This scenario is more feasible when the password check is more time consuming and we don't do it if the username doesn't match. However, it's better to prevent that situation by fixing this.

Really liked the example of `subtle.ConstantTimeCompare`. If this is reconsidered again, a look should be taken to that implementation.

Thanks for the review <3